diff --git a/lib/ReactStaticPackager.js b/lib/ReactStaticPackager.js
index 551b6071819ea7c8c28106281793841d63791856..1556a6abc7a38fc1959eec279810a14614c08772 100644
--- a/lib/ReactStaticPackager.js
+++ b/lib/ReactStaticPackager.js
@@ -291,8 +291,9 @@ async function loadBundleUncached(bundle, bundleGraph, getInlineBundleContents)
             packagingBundles.set(entryBundle, getInlineBundleContents(entryBundle, bundleGraph));
           }
           let packagedBundle = await (0, _nullthrows().default)(packagingBundles.get(entryBundle));
+          let inlineType = (0, _nullthrows().default)(entryBundle.getMainEntry()).meta.inlineType;
           let contents = await (0, _utils().blobToString)(packagedBundle.contents);
-          contents = `module.exports = ${contents}`;
+          contents = `module.exports = ${inlineType === 'string' ? JSON.stringify(contents) : contents}`;
           return [[entryBundle.id, [entryBundle, (0, _nullthrows().default)(entryBundle.getMainEntry()), contents]]];
         });
       } else if ((entryBundle === null || entryBundle === void 0 ? void 0 : entryBundle.type) === 'js') {
@@ -597,7 +597,7 @@ function runModule(code, filename, id, require, parcelRequire) {
   return module.exports;
 }
 function getCacheKey(asset) {
-  return asset.filePath + '#' + asset.env.context + (asset.uniqueKey ? '-' + asset.uniqueKey : '');
+  return (asset.pipeline ? asset.pipeline : '') + asset.filePath + '#' + asset.env.context + (asset.uniqueKey ? '-' + asset.uniqueKey : '');
 }
 function getSpecifier(dep) {
   if (typeof dep.meta.placeholder === 'string') {
