diff --git a/lib/client.d.ts b/lib/client.d.ts
index abfc3ae8bc4cee00f322501e209bc8a63ee3a1d1..cfe5e732732d857804b87ec2534e90aea6749330 100644
--- a/lib/client.d.ts
+++ b/lib/client.d.ts
@@ -1,6 +1,5 @@
 /* @jsxRuntime automatic */ import { ReactNode } from 'react';
 import { HydrationOptions } from 'react-dom/client';
-type CallServerCallback = <T>(id: string, args: any[]) => Promise<T>;
 export declare function setServerCallback(cb: CallServerCallback): void;
 export type CallServerCallback = <T>(id: string, args: any[]) => Promise<T>;
 export interface HydrateOptions extends HydrationOptions {
