diff --git a/lib/TSModuleGraph.js b/lib/TSModuleGraph.js
index 78d0e8f37f5535ab46061dbfc07b5986fa848cfa..2cea4cffdeb4cc626b547b9c704e4f49e0a60be1 100644
--- a/lib/TSModuleGraph.js
+++ b/lib/TSModuleGraph.js
@@ -138,7 +138,7 @@ class TSModuleGraph {
     for (let e of module.exports) {
       if (e.name === name) {
         return this.getExport(module, e);
-      } else if (e.specifier) {
+      } else if (e.specifier && this.getModule(e.specifier)) {
         const m = this.resolveExport((0, _nullthrows().default)(this.getModule(e.specifier)), name);
         if (m) {
           return m;
@@ -154,7 +154,7 @@ class TSModuleGraph {
         if (exp) {
           res.push(exp);
         }
-      } else if (e.specifier) {
+      } else if (e.specifier && this.getModule(e.specifier)) {
         let m = this.getModule(e.specifier);
         if (m) {
           res.push(...this.getAllExports(m, true));
