diff --git a/lib/JSTransformer.js b/lib/JSTransformer.js
index c2a45a9665b4b88836d720669b997872082bb143..f46b0edf814e398ed2a2584efeb040c4b77d58db 100644
--- a/lib/JSTransformer.js
+++ b/lib/JSTransformer.js
@@ -447,14 +447,8 @@ var _default = exports.default = new (_plugin().Transformer)({
               },
               loc: {
                 filePath: asset.filePath,
-                start: {
-                  line: loc.start_line + Number(asset.meta.startLine ?? 1) - 1,
-                  column: loc.start_col
-                },
-                end: {
-                  line: loc.end_line + Number(asset.meta.startLine ?? 1) - 1,
-                  column: loc.end_col
-                }
+                line: loc.line,
+                column: loc.col
               },
               invalidateOnFileChange(filePath) {
                 asset.invalidateOnFileChange(filePath);
