diff --git a/lib/JSTransformer.js b/lib/JSTransformer.js
index 1e0991fc8f9c231b70eefd7e177119a06162eeca..e5be5a0901ebc97cbbbb7bc0c774787de36fa2b5 100644
--- a/lib/JSTransformer.js
+++ b/lib/JSTransformer.js
@@ -446,6 +446,11 @@ var _default = exports.default = new (_plugin().Transformer)({
                   specifierType: 'esm'
                 });
               },
+              loc: {
+                filePath: asset.filePath,
+                line: loc.line,
+                col: loc.col
+              },
               invalidateOnFileChange(filePath) {
                 asset.invalidateOnFileChange(filePath);
               },
