diff --git a/dist/index.d.ts b/dist/index.d.ts
index 4d065a3eb77d7dd73743a62bca53aa0ad42a43e1..e2d4c50c55fd1810742017cfe50083da4a7c3cac 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -2,7 +2,7 @@ import * as _storybook_types from '@storybook/types';
 import { Addon_ClientStoryApi, Addon_Loadable, Args, ComponentAnnotations, AnnotatedStoryFn, ArgsStoryFn, ArgsFromMeta, StoryAnnotations, DecoratorFunction, StrictArgs, LoaderFunction, StoryContext as StoryContext$1, ProjectAnnotations, StoryAnnotationsOrFn, Store_CSFExports, StoriesWithPartialProps } from '@storybook/types';
 export { ArgTypes, Args, Parameters, StrictArgs } from '@storybook/types';
 import { R as ReactRenderer } from './types-0fc72a6d.js';
-import { ComponentType, ComponentProps, JSXElementConstructor } from 'react';
+import { ComponentType, ComponentProps, JSXElementConstructor, JSX } from 'react';
 import { Simplify, SetOptional } from 'type-fest';
 
 interface ClientApi extends Addon_ClientStoryApi<ReactRenderer['storyResult']> {
