diff --git a/lib/rules/no-relative-import.js b/lib/rules/no-relative-import.js
index 1298280d1e67e7753fc8bc7d9fc8dcbf1d8477dd..069e1a8fd771bd91f796fa5ec830e6d0ede4a709 100644
--- a/lib/rules/no-relative-import.js
+++ b/lib/rules/no-relative-import.js
@@ -19,10 +19,6 @@ var _pathIsInside = require('path-is-inside');
 
 var _pathIsInside2 = _interopRequireDefault(_pathIsInside);
 
-var _minimatch = require('minimatch');
-
-var _minimatch2 = _interopRequireDefault(_minimatch);
-
 var _path = require('path');
 
 var _path2 = _interopRequireDefault(_path);
@@ -40,12 +36,15 @@ var meta = exports.meta = {
 
 var create = exports.create = function create(context) {
   var _context$options = _slicedToArray(context.options, 1),
-      moduleUtilOptions = _context$options[0];
+    moduleUtilOptions = _context$options[0];
 
   var sourceFsPath = context.getFilename();
   var packages = (0, _getMonorepoPackages2.default)(process.cwd());
 
   return (0, _moduleVisitor2.default)(function (node) {
+    if (!node.value.includes('..')) {
+      return;
+    }
     var resolvedPath = (0, _resolve2.default)(node.value, context);
     var packageDir = getPackageDir(sourceFsPath, packages);
 
@@ -73,7 +72,7 @@ var create = exports.create = function create(context) {
 
 var getPackageDir = function getPackageDir(filePath, packages) {
   var match = packages.find(function (pkg) {
-    return (0, _minimatch2.default)(filePath, _path2.default.join(pkg.location, '**'));
+    return (0, _pathIsInside2.default)(filePath, pkg.location);
   });
   if (match) {
     return match.location;
