diff --git a/index.js b/index.js
index aacacb1da77e81537385fba58140be861451f51a..1cb51789b4ac19940311500ed1fcb0c0e8317ffe 100644
--- a/index.js
+++ b/index.js
@@ -29,7 +29,7 @@ const findPackages = (packageSpecs, rootDirectory) => {
     .filter(({ package: { name } = {} }) => name);
 };
 
-const getPackages = directory => {
+const getPackagesNew = (directory) => {
   const lernaJsonPath = path.join(directory, 'lerna.json');
   if (fs.existsSync(lernaJsonPath)) {
     const lernaJson = loadJsonFile.sync(lernaJsonPath);
@@ -58,6 +58,20 @@ const getPackages = directory => {
 
   // Bail if we don't find any packages
   return [];
+}
+
+const packageMap = new Map();
+
+const getPackages = directory => {
+  // We assume that the set of monorepo packages for this directory will
+  // not change during a single eslint run
+  let packages = packageMap.get(directory);
+  if (!packages) {
+    packages = getPackagesNew(directory);
+    packageMap.set(directory, packages);
+  }
+
+  return packages;
 };
 
 module.exports = getPackages;
