diff --git a/src/nwsapi.js b/src/nwsapi.js
index 872026b4ab3462f3c2411bc564dd428bd3165323..f71ae753743c9d6155e2f522c1ac3c7fdbb05c32 100644
--- a/src/nwsapi.js
+++ b/src/nwsapi.js
@@ -1241,9 +1241,10 @@
                     'if((e===n||e.autofocus)){' + source + '}';
                   break;
                 case 'focus-within':
-                  source = 'if(n=s.isFocusable(e)){' +
-                    'if(n!==e){while(n){n=n.parentElement;if(n===e)break;}}}' +
-                    'if((n===e||n.autofocus)){' + source + '}';
+                  // Check if e contains the activeElement by walking up from activeElement
+                  source = 'if(s.doc.hasFocus()){' +
+                    'n=s.doc.activeElement;' +
+                    'while(n){if(n===e){' + source + 'break;}n=n.parentElement;}}';
                   break;
                 default:
                   emit('\'' + expression + '\'' + qsInvalid);
