diff --git a/preset.js b/preset.js
index b2076be6b23131344605501f218ce8bda0d00ba0..f3649400a7232af59aca9797da438df12cfbbb97 100644
--- a/preset.js
+++ b/preset.js
@@ -119,10 +119,12 @@ async function createParcel(options, isDev = false) {
         resolveFrom: __filename,
       },
     ],
-    defaultTargetOptions: {
-      distDir: options.outputDir,
-      publicUrl: "./",
-      shouldScopeHoist: isDev ? false : true,
-    },
+    targets: {
+      storybook: {
+          distDir: options.outputDir,
+            publicUrl: "./",
+            scopeHoist: isDev ? false : true,
+          }
+    }
   });
 }
