diff --git a/node_modules/@jest/types/node_modules/chalk/index.d.ts b/node_modules/@jest/types/node_modules/chalk/index.d.ts
index 7e22c45..21b7ec4 100644
--- a/node_modules/@jest/types/node_modules/chalk/index.d.ts
+++ b/node_modules/@jest/types/node_modules/chalk/index.d.ts
@@ -400,6 +400,7 @@ This simply means that `chalk.red.yellow.green` is equivalent to `chalk.green`.
 */
 declare const chalk: chalk.Chalk & chalk.ChalkFunction & {
 	supportsColor: chalk.ColorSupport | false;
+	// @ts-ignore
 	Level: typeof LevelEnum;
 	Color: Color;
 	ForegroundColor: ForegroundColor;
