diff --git a/node_modules/@parcel/transformer-css/lib/CSSTransformer.js b/node_modules/@parcel/transformer-css/lib/CSSTransformer.js
index b8dd9f2..0bad8ab 100644
--- a/node_modules/@parcel/transformer-css/lib/CSSTransformer.js
+++ b/node_modules/@parcel/transformer-css/lib/CSSTransformer.js
@@ -88,6 +88,9 @@ var _default = exports.default = new (_plugin().Transformer)({
     } else if (Array.isArray(contents === null || contents === void 0 || (_contents$cssModules4 = contents.cssModules) === null || _contents$cssModules4 === void 0 ? void 0 : _contents$cssModules4.exclude)) {
       contents.cssModules.exclude = contents.cssModules.exclude.map(exclude => typeof exclude === 'string' ? (0, _utils().globToRegex)(exclude) : exclude);
     }
+    if (config.searchPath.endsWith('home.global.css')) {
+      contents = {...contents, pseudoClasses: null};
+    }
     return contents;
   },
   async transform({
