diff --git a/node_modules/@parcel/transformer-postcss/lib/PostCSSTransformer.js b/node_modules/@parcel/transformer-postcss/lib/PostCSSTransformer.js
index 3a35814..ee06412 100644
--- a/node_modules/@parcel/transformer-postcss/lib/PostCSSTransformer.js
+++ b/node_modules/@parcel/transformer-postcss/lib/PostCSSTransformer.js
@@ -96,10 +96,10 @@ var _default = new (_plugin().Transformer)({
     const postcss = await loadPostcss(options, asset.filePath);
     return {
       type: 'postcss',
-      version: '8.2.1',
+      version: '7.0.0',
       program: postcss.parse(await asset.getCode(), {
         from: asset.filePath
-      }).toJSON()
+      })
     };
   },
   async transform({
@@ -131,10 +131,10 @@ var _default = new (_plugin().Transformer)({
     }
     const postcss = await loadPostcss(options, asset.filePath);
     let ast = (0, _nullthrows().default)(await asset.getAST());
-    let program = postcss.fromJSON(ast.program);
+    let program = ast.program;
     let plugins = [...config.hydrated.plugins];
     let cssModules = null;
-    if (config.hydrated.modules) {
+    if (config.hydrated.modules && asset.meta.type !== 'tag') {
       asset.meta.cssModulesCompiled = 'postcss';
       let code = asset.isASTDirty() ? null : await asset.getCode();
       if (Object.keys(config.hydrated.modules).length === 0 && code && !isLegacy && !LEGACY_MODULE_RE.test(code)) {
@@ -198,7 +198,7 @@ var _default = new (_plugin().Transformer)({
 
       // TODO: should this be resolved from the project root?
       let postcssModules = await options.packageManager.require('postcss-modules', asset.filePath, {
-        range: '^4.3.0',
+        range: '^3.0.0',
         saveDev: true,
         shouldAutoInstall: options.shouldAutoInstall
       });
@@ -241,8 +241,8 @@ var _default = new (_plugin().Transformer)({
     } = await postcss(plugins).process(program, config.hydrated);
     asset.setAST({
       type: 'postcss',
-      version: '8.2.1',
-      program: root.toJSON()
+      version: '7.0.0',
+      program: root
     });
     for (let msg of messages) {
       if (msg.type === 'dependency') {
@@ -276,11 +276,11 @@ var _default = new (_plugin().Transformer)({
         // This syntax enables shaking the invidual statements, so that unused classes don't even exist in JS.
         ([className, classNameHashed]) => `module.exports[${JSON.stringify(className)}] = ${JSON.stringify(classNameHashed)};`).join('\n');
       }
-      asset.symbols.ensure();
-      for (let [k, v] of cssModulesList) {
-        asset.symbols.set(k, v);
-      }
-      asset.symbols.set('default', 'default');
+      // asset.symbols.ensure();
+      // for (let [k, v] of cssModulesList) {
+      //   asset.symbols.set(k, v);
+      // }
+      // asset.symbols.set('default', 'default');
       assets.push({
         type: 'js',
         content: code
@@ -295,7 +295,7 @@ var _default = new (_plugin().Transformer)({
   }) {
     const postcss = await loadPostcss(options, asset.filePath);
     let code = '';
-    postcss.stringify(postcss.fromJSON(ast.program), c => {
+    postcss.stringify(ast.program, c => {
       code += c;
     });
     return {
diff --git a/node_modules/@parcel/transformer-postcss/lib/constants.js b/node_modules/@parcel/transformer-postcss/lib/constants.js
index a16ee4c..995a367 100644
--- a/node_modules/@parcel/transformer-postcss/lib/constants.js
+++ b/node_modules/@parcel/transformer-postcss/lib/constants.js
@@ -4,5 +4,5 @@ Object.defineProperty(exports, "__esModule", {
   value: true
 });
 exports.POSTCSS_RANGE = void 0;
-const POSTCSS_RANGE = '^8.2.1';
+const POSTCSS_RANGE = '^7.0.0';
 exports.POSTCSS_RANGE = POSTCSS_RANGE;
\ No newline at end of file
