diff --git a/node_modules/@parcel/transformer-postcss/lib/PostCSSTransformer.js b/node_modules/@parcel/transformer-postcss/lib/PostCSSTransformer.js
index a989028..6924722 100644
--- a/node_modules/@parcel/transformer-postcss/lib/PostCSSTransformer.js
+++ b/node_modules/@parcel/transformer-postcss/lib/PostCSSTransformer.js
@@ -116,7 +116,7 @@ var _default = new (_plugin().Transformer)({
   canReuseAST({
     ast
   }) {
-    return ast.type === 'postcss' && _semver().default.satisfies(ast.version, '^8.2.1');
+    return ast.type === 'postcss' && _semver().default.satisfies(ast.version, '^7.0.0');
   },
 
   async parse({
@@ -131,10 +131,10 @@ var _default = new (_plugin().Transformer)({
     const postcss = await loadPostcss(options, asset.filePath);
     return {
       type: 'postcss',
-      version: '8.2.1',
+      version: '7.0.0',
       program: postcss.parse(await asset.getCode(), {
         from: asset.filePath
-      }).toJSON()
+      })
     };
   },
 
@@ -164,7 +164,7 @@ var _default = new (_plugin().Transformer)({
     }
 
     let ast = (0, _nullthrows().default)(await asset.getAST());
-    let program = postcss.fromJSON(ast.program);
+    let program = ast.program;
     let code = asset.isASTDirty() ? null : await asset.getCode();
 
     if (code == null || COMPOSES_RE.test(code)) {
@@ -199,8 +199,8 @@ var _default = new (_plugin().Transformer)({
     } = await postcss(plugins).process(program, config.hydrated);
     asset.setAST({
       type: 'postcss',
-      version: '8.2.1',
-      program: root.toJSON()
+      version: '7.0.0',
+      program: root
     });
 
     for (let msg of messages) {
@@ -226,13 +226,13 @@ var _default = new (_plugin().Transformer)({
         ([className, classNameHashed]) => `module.exports[${JSON.stringify(className)}] = ${JSON.stringify(classNameHashed)};`).join('\n');
       }
 
-      asset.symbols.ensure();
+      // asset.symbols.ensure();
 
-      for (let [k, v] of cssModulesList) {
-        asset.symbols.set(k, v);
-      }
+      // for (let [k, v] of cssModulesList) {
+      //   asset.symbols.set(k, v);
+      // }
 
-      asset.symbols.set('default', 'default');
+      // asset.symbols.set('default', 'default');
       assets.push({
         type: 'js',
         filePath: asset.filePath + '.js',
@@ -250,7 +250,7 @@ var _default = new (_plugin().Transformer)({
   }) {
     const postcss = await loadPostcss(options, asset.filePath);
     let code = '';
-    postcss.stringify(postcss.fromJSON(ast.program), c => {
+    postcss.stringify(ast.program, c => {
       code += c;
     });
     return {
@@ -294,7 +294,7 @@ function createLoader(asset, resolve) {
 
 function loadPostcss(options, from) {
   return options.packageManager.require('postcss', from, {
-    range: '^8.2.1',
+    range: '^7.0.0',
     saveDev: true,
     shouldAutoInstall: options.shouldAutoInstall
   });
