diff --git a/node_modules/@parcel/bundler-default/lib/DefaultBundler.js b/node_modules/@parcel/bundler-default/lib/DefaultBundler.js
index 6f2b7a1..6de3ecf 100644
--- a/node_modules/@parcel/bundler-default/lib/DefaultBundler.js
+++ b/node_modules/@parcel/bundler-default/lib/DefaultBundler.js
@@ -380,7 +380,7 @@ function createIdealGraph(assetGraph, config, entries) {
              * and bundles of other types should be merged in the next step
              */
             let bundleGroupRootAsset = (0, _nullthrows().default)(bundleGroup.mainEntryAsset);
-            if (entries.has(bundleGroupRootAsset) && canMerge(bundleGroupRootAsset, childAsset) && dependency.bundleBehavior == null) {
+            if (parentAsset.type !== childAsset.type && entries.has(bundleGroupRootAsset) && canMerge(bundleGroupRootAsset, childAsset) && dependency.bundleBehavior == null) {
               bundleId = bundleGroupNodeId;
             }
             if (bundleId == null) {
