diff --git a/node_modules/@parcel/packager-js/lib/ScopeHoistingPackager.js b/node_modules/@parcel/packager-js/lib/ScopeHoistingPackager.js
index 6e85021..fc81874 100644
--- a/node_modules/@parcel/packager-js/lib/ScopeHoistingPackager.js
+++ b/node_modules/@parcel/packager-js/lib/ScopeHoistingPackager.js
@@ -395,7 +395,7 @@ class ScopeHoistingPackager {
     }
 
     let [depMap, replacements] = this.buildReplacements(asset, deps);
-    let [prepend, prependLines, append] = this.buildAssetPrelude(asset, deps);
+    let [prepend, prependLines, append] = this.buildAssetPrelude(asset, deps, replacements);
 
     if (prependLines > 0) {
       sourceMap === null || sourceMap === void 0 ? void 0 : sourceMap.offsetLines(1, prependLines);
@@ -762,7 +762,7 @@ ${code}
     return [res, lineCount];
   }
 
-  buildAssetPrelude(asset, deps) {
+  buildAssetPrelude(asset, deps, replacements) {
     let prepend = '';
     let prependLineCount = 0;
     let append = '';
@@ -834,6 +834,7 @@ ${code}
         // additional assignments after each mutation of the original binding.
         prepend += `\n${usedExports.map(exp => {
           let resolved = this.getSymbolResolution(asset, asset, exp);
+          resolved = replacements.get(resolved) || resolved;
           let get = this.buildFunctionExpression([], resolved);
           let set = asset.meta.hasCJSExports ? ', ' + this.buildFunctionExpression(['v'], `${resolved} = v`) : '';
           return `$parcel$export($${assetId}$exports, ${JSON.stringify(exp)}, ${get}${set});`;
