diff --git a/node_modules/@testing-library/user-event/dist/type.js b/node_modules/@testing-library/user-event/dist/type.js
index fd7d7b9..6494d98 100644
--- a/node_modules/@testing-library/user-event/dist/type.js
+++ b/node_modules/@testing-library/user-event/dist/type.js
@@ -304,6 +304,7 @@ function typeCharacter(char, {
       key,
       keyCode,
       charCode: keyCode,
+      which: keyCode,
       ...eventOverrides
     });
 
