diff --git a/node_modules/recast/lib/printer.js b/node_modules/recast/lib/printer.js
index 9bedd42..33941b0 100644
--- a/node_modules/recast/lib/printer.js
+++ b/node_modules/recast/lib/printer.js
@@ -2199,7 +2199,7 @@ function printFunctionParams(path, options, print) {
 function maybePrintImportAssertions(path, options, print) {
     var n = path.getValue();
     if (n.assertions && n.assertions.length > 0) {
-        var parts = [" assert {"];
+        var parts = [" with {"];
         var printed = path.map(print, "assertions");
         var flat = (0, lines_1.fromString)(", ").join(printed);
         if (flat.length > 1 || flat.getLineLength(1) > options.wrapColumn) {
