diff --git a/node_modules/storybook-builder-parcel/preset.js b/node_modules/storybook-builder-parcel/preset.js
index b2076be..7d949d4 100644
--- a/node_modules/storybook-builder-parcel/preset.js
+++ b/node_modules/storybook-builder-parcel/preset.js
@@ -119,10 +119,15 @@ async function createParcel(options, isDev = false) {
         resolveFrom: __filename,
       },
     ],
-    defaultTargetOptions: {
-      distDir: options.outputDir,
-      publicUrl: "./",
-      shouldScopeHoist: isDev ? false : true,
-    },
+    targets: {
+      storybook: {
+        distDir: options.outputDir,
+        publicUrl: "./",
+        scopeHoist: isDev ? false : true,
+        engines: {
+          browsers: ['last 2 Chrome version', 'last 2 Safari versions', 'last 2 Edge version', 'last 2 Firefox versions']
+        }
+      }
+    }
   });
 }
